ide/che-core-ide-stacks/src/main/resources/stacks.json

```diff
@@ -2771,5 +2771,50 @@
     "name": "type-blank.svg",
     "mediaType": "image/svg+xml"
   }
- }
+  },
+  {
+    "id": "openshift",
+    "name": "openshift",
+    "description": "Openshift Workspace",
+    "scope": "general",
+    "tags": [
+      "Openshift"
+    ],
+    "workspaceConfig": {
+      "environments": {
+        "default": {
+          "machines": {
+            "dev-machine/main": {
+              "installers": [
+                "org.eclipse.che.exec", "org.eclipse.che.terminal", "org.eclipse.che.ws-agent"
+              ],
+              "servers": {},
+              "attributes" : {
+                "memoryLimitBytes": "2147483648"
+              }
+            }
+          },
+          "recipe": {
```

Comment:
To make the things clear - the format is not "dedicated", it is "native" for K8s, OS, which I consider as a huge advantage of Che6 SPI.
Showing ability to support it is one of the main goal of initial implementation.
The problem you are talking about can be solved with more than one way:
- defining another environment, i.e. one for Docker, another for OSIO (we can imagine some more for example for Local computer, no docker, no containers), so one or another will work depending on available infrastructure
- adding one more infrastructure type which works on OS but support some other, simplified format (docker image, dockerfile etc)
- extending this infrastructure implementation making it support more than one recipe type (like in Docker impl) - possible by design but I personally would not prefer this way 

And, of course they can be combined.